### DISCUSSION TITLE
turn by turn crash

### DISCUSSION THREAD

**Discussion by alihelmy27** [Imperial-lord/mapbox-flutter#18] [Q&A]
whenever I press the button to start navigating and open the turn by turn screen the app crashes and stops.

**Comment by alihelmy27** [comment]
E/libc++abi( 6511): terminating with uncaught exception of type jni::PendingJavaException
F/libc    ( 6511): Fatal signal 6 (SIGABRT), code -1 (SI_QUEUE) in tid 6511 (pbox_navigation), pid 6511 (pbox_navigation)
*** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
Build fingerprint: 'Xiaomi/platina/platina:10/QKQ1.190910.002/V12.0.3.0.QDTMIXM:user/release-keys'
Revision: '0'
ABI: 'arm64'
Timestamp: 2022-06-05 18:27:09+0200
pid: 6511, tid: 6511, name: pbox_navigation  >>> com.example.mapbox_navigation <<<
uid: 10303
signal 6 (SIGABRT), code -1 (SI_QUEUE), fault addr --------
Abort message: 'terminating with uncaught exception of type jni::PendingJavaException'
    x0  0000000000000000  x1  000000000000196f  x2  0000000000000006  x3  0000007ff9d95d90
    x4  85b0ff6d6e68736f  x5  85b0ff6d6e68736f  x6  85b0ff6d6e68736f  x7  ffff7f7f7f7f7f7f
    x8  00000000000000f0  x9  86b2f27b832b70b9  x10 0000000000000001  x11 0000000000000000
    x12 fffffff0fffffbdf  x13 00000000629cd95d  x14 0026422e2589dc63  x15 00007731ea7f3cb6
    x16 000000731c47a8c0  x17 000000731c456900  x18 000000731e588000  x19 000000000000196f
    x20 000000000000196f  x21 00000000ffffffff  x22 ffffff80ffffffc8  x23 0000007ff9d95fe0
    x24 0000007ff9d95ec0  x25 0000007ff9d95f00  x26 0000007207fdf968  x27 0000007207fdf960
    x28 0000007207fdf980  x29 0000007ff9d95e30
    sp  0000007ff9d95d70  lr  000000731c4080c4  pc  000000731c4080f0
backtrace:
      #00 pc 00000000000830f0  /apex/com.android.runtime/lib64/bionic/libc.so (abort+160) (BuildId: 328a3518945f3e056fc055f1eea1cc35)
      #01 pc 0000000000373aa4  /data/app/com.example.mapbox_navigation-iPsHJSLW9YJhyTET_AylWQ==/lib/arm64/libmapbox-gl.so (BuildId: 5340fab4c92fd87f8304775de4bc95627e863fd7)
      #02 pc 0000000000373c14  /data/app/com.example.mapbox_navigation-iPsHJSLW9YJhyTET_AylWQ==/lib/arm64/libmapbox-gl.so (BuildId: 5340fab4c92fd87f8304775de4bc95627e863fd7)
      #03 pc 00000000003711d4  /data/app/com.example.mapbox_navigation-iPsHJSLW9YJhyTET_AylWQ==/lib/arm64/libmapbox-gl.so (BuildId: 5340fab4c92fd87f8304775de4bc95627e863fd7)
      #04 pc 0000000000370800  /data/app/com.example.mapbox_navigation-iPsHJSLW9YJhyTET_AylWQ==/lib/arm64/libmapbox-gl.so (BuildId: 5340fab4c92fd87f8304775de4bc95627e863fd7)
      #05 pc 000000000037075c  /data/app/com.example.mapbox_navigation-iPsHJSLW9YJhyTET_AylWQ==/lib/arm64/libmapbox-gl.so (__cxa_throw+112) (BuildId: 5340fab4c92fd87f8304775de4bc95627e863fd7)
      #06 pc 0000000000094680  /data/app/com.example.mapbox_navigation-iPsHJSLW9YJhyTET_AylWQ==/lib/arm64/libmapbox-gl.so (BuildId: 5340fab4c92fd87f8304775de4bc95627e863fd7)
      #07 pc 00000000000aee88  /data/app/com.example.mapbox_navigation-iPsHJSLW9YJhyTET_AylWQ==/lib/arm64/libmapbox-gl.so (BuildId: 5340fab4c92fd87f8304775de4bc95627e863fd7)
      #08 pc 0000000000115b8c  /data/app/com.example.mapbox_navigation-iPsHJSLW9YJhyTET_AylWQ==/lib/arm64/libmapbox-gl.so (BuildId: 5340fab4c92fd87f8304775de4bc95627e863fd7)
      #09 pc 00000000001c3898  /data/app/com.example.mapbox_navigation-iPsHJSLW9YJhyTET_AylWQ==/lib/arm64/libmapbox-gl.so (BuildId: 5340fab4c92fd87f8304775de4bc95627e863fd7)
      #10 pc 00000000001c3314  /data/app/com.example.mapbox_navigation-iPsHJSLW9YJhyTET_AylWQ==/lib/arm64/libmapbox-gl.so (BuildId: 5340fab4c92fd87f8304775de4bc95627e863fd7)
      #11 pc 00000000001c5b70  /data/app/com.example.mapbox_navigation-iPsHJSLW9YJhyTET_AylWQ==/lib/arm64/libmapbox-gl.so (BuildId: 5340fab4c92fd87f8304775de4bc95627e863fd7)
      #12 pc 000000000022177c  /data/app/com.example.mapbox_navigation-iPsHJSLW9YJhyTET_AylWQ==/lib/arm64/libmapbox-gl.so (BuildId: 5340fab4c92fd87f8304775de4bc95627e863fd7)
      #13 pc 00000000000fb3e0  /data/app/com.example.mapbox_navigation-iPsHJSLW9YJhyTET_AylWQ==/lib/arm64/libmapbox-gl.so (BuildId: 5340fab4c92fd87f8304775de4bc95627e863fd7)
      #14 pc 000000000021b0c4  /data/app/com.example.mapbox_navigation-iPsHJSLW9YJhyTET_AylWQ==/lib/arm64/libmapbox-gl.so (BuildId: 5340fab4c92fd87f8304775de4bc95627e863fd7)
      #15 pc 000000000021a930  /data/app/com.example.mapbox_navigation-iPsHJSLW9YJhyTET_AylWQ==/lib/arm64/libmapbox-gl.so (BuildId: 5340fab4c92fd87f8304775de4bc95627e863fd7)
      #16 pc 000000000021a9fc  /data/app/com.example.mapbox_navigation-iPsHJSLW9YJhyTET_AylWQ==/lib/arm64/libmapbox-gl.so (BuildId: 5340fab4c92fd87f8304775de4bc95627e863fd7)
      #17 pc 0000000000219c24  /data/app/com.example.mapbox_navigation-iPsHJSLW9YJhyTET_AylWQ==/lib/arm64/libmapbox-gl.so (BuildId: 5340fab4c92fd87f8304775de4bc95627e863fd7)
      #18 pc 00000000000183a0  /system/lib64/libutils.so (android::Looper::pollInner(int)+832) (BuildId: f52e94e6224d7e15faba92ee0326b191)
      #19 pc 0000000000017fc0  /system/lib64/libutils.so (android::Looper::pollOnce(int, int*, int*, void**)+56) (BuildId: f52e94e6224d7e15faba92ee0326b191)
      #20 pc 000000000013eca8  /system/lib64/libandroid_runtime.so (android::android_os_MessageQueue_nativePollOnce(_JNIEnv*, _jobject*, long, int)+44) (BuildId: 58119cd499c9126f89ff7f6888912397)
      #21 pc 000000000141a84c  /system/framework/arm64/boot-framework.oat (art_jni_trampoline+140) (BuildId: 7f789458715e814d322792a9aa87472ab8caa0b8)
      #22 pc 000000000200d434  /memfd:/jit-cache (deleted) (android.os.MessageQueue.next+180)
      #23 pc 0000000002047634  /memfd:/jit-cache (deleted) (android.os.Looper.loop+404)
      #24 pc 000000000013763c  /apex/com.android.runtime/lib64/libart.so (art_quick_osr_stub+60) (BuildId: a64ec0d2e59c0dc84e59dad386f412d8)
      #25 pc 00000000003380f8  /apex/com.android.runtime/lib64/libart.so (art::jit::Jit::MaybeDoOnStackReplacement(art::Thread*, art::ArtMethod*, unsigned int, int, art::JValue*)+1688) (BuildId: a64ec0d2e59c0dc84e59dad386f412d8)
      #26 pc 00000000005abc84  /apex/com.android.runtime/lib64/libart.so (MterpMaybeDoOnStackReplacement+212) (BuildId: a64ec0d2e59c0dc84e59dad386f412d8)
      #27 pc 0000000000136350  /apex/com.android.runtime/lib64/libart.so (MterpHelpers+240) (BuildId: a64ec0d2e59c0dc84e59dad386f412d8)
      #28 pc 0000000000347b90  /system/framework/framework.jar (android.os.Looper.loop+1124)
      #29 pc 00000000005a3c38  /apex/com.android.runtime/lib64/libart.so (MterpInvokeStatic+1040) (BuildId: a64ec0d2e59c0dc84e59dad386f412d8)
      #30 pc 0000000000131994  /apex/com.android.runtime/lib64/libart.so (mterp_op_invoke_static+20) (BuildId: a64ec0d2e59c0dc84e59dad386f412d8)
      #31 pc 00000000001a72e8  /system/framework/framework.jar (android.app.ActivityThread.main+196)
      #32 pc 00000000002b4b14  /apex/com.android.runtime/lib64/libart.so (_ZN3art11interpreterL7ExecuteEPNS_6ThreadERKNS_20CodeItemDataAccessorERNS_11ShadowFrameENS_6JValueEbb.llvm.17460956533834400288+240) (BuildId: a64ec0d2e59c0dc84e59dad386f412d8)
      #33 pc 0000000000592430  /apex/com.android.runtime/lib64/libart.so (artQuickToInterpreterBridge+1032) (BuildId: a64ec0d2e59c0dc84e59dad386f412d8)
      #34 pc 0000000000140468  /apex/com.android.runtime/lib64/libart.so (art_quick_to_interpreter_bridge+88) (BuildId: a64ec0d2e59c0dc84e59dad386f412d8)
      #35 pc 00000000001375b8  /apex/com.android.runtime/lib64/libart.so (art_quick_invoke_static_stub+568) (BuildId: a64ec0d2e59c0dc84e59dad386f412d8)
      #36 pc 000000000014600c  /apex/com.android.runtime/lib64/libart.so (art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+276) (BuildId: a64ec0d2e59c0dc84e59dad386f412d8)
      #37 pc 00000000004b0e34  /apex/com.android.runtime/lib64/libart.so (art::(anonymous namespace)::InvokeWithArgArray(art::ScopedObjectAccessAlreadyRunnable const&, art::ArtMethod*, art::(anonymous namespace)::ArgArray*, art::JValue*, char const*)+104) (BuildId: a64ec0d2e59c0dc84e59dad386f412d8)
      #38 pc 00000000004b29d8  /apex/com.android.runtime/lib64/libart.so (art::InvokeMethod(art::ScopedObjectAccessAlreadyRunnable const&, _jobject*, _jobject*, _jobject*, unsigned long)+1480) (BuildId: a64ec0d2e59c0dc84e59dad386f412d8)
      #39 pc 000000000043dc70  /apex/com.android.runtime/lib64/libart.so (art::Method_invoke(_JNIEnv*, _jobject*, _jobject*, _jobjectArray*)+52) (BuildId: a64ec0d2e59c0dc84e59dad386f412d8)
      #40 pc 00000000006d4ea4  /system/framework/arm64/boot.oat (art_jni_trampoline+180) (BuildId: c5d52f7d0582c5beca43ae07fe5d45b1921b2572)
      #41 pc 0000000000137334  /apex/com.android.runtime/lib64/libart.so (art_quick_invoke_stub+548) (BuildId: a64ec0d2e59c0dc84e59dad386f412d8)
      #42 pc 0000000000145fec  /apex/com.android.runtime/lib64/libart.so (art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+244) (BuildId: a64ec0d2e59c0dc84e59dad386f412d8)
      #43 pc 00000000002e3800  /apex/com.android.runtime/lib64/libart.so (art::interpreter::ArtInterpreterToCompiledCodeBridge(art::Thread*, art::ArtMethod*, art::ShadowFrame*, unsigned short, art::JValue*)+384) (BuildId: a64ec0d2e59c0dc84e59dad386f412d8)
      #44 pc 00000000002dea60  /apex/com.android.runtime/lib64/libart.so (bool art::interpreter::DoCall<false, false>(art::ArtMethod*, art::Thread*, art::ShadowFrame&, art::Instruction const*, unsigned short, art::JValue*)+892) (BuildId: a64ec0d2e59c0dc84e59dad386f412d8)
      #45 pc 00000000005a0ec8  /apex/com.android.runtime/lib64/libart.so (MterpInvokeVirtual+648) (BuildId: a64ec0d2e59c0dc84e59dad386f412d8)
      #46 pc 0000000000131814  /apex/com.android.runtime/lib64/libart.so (mterp_op_invoke_virtual+20) (BuildId: a64ec0d2e59c0dc84e59dad386f412d8)
      #47 pc 000000000039fdc2  /system/framework/framework.jar (com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run+22)
      #48 pc 00000000002b4b14  /apex/com.android.runtime/lib64/libart.so (_ZN3art11interpreterL7ExecuteEPNS_6ThreadERKNS_20CodeItemDataAccessorERNS_11ShadowFrameENS_6JValueEbb.llvm.17460956533834400288+240) (BuildId: a64ec0d2e59c0dc84e59dad386f412d8)
      #49 pc 0000000000592430  /apex/com.android.runtime/lib64/libart.so (artQuickToInterpreterBridge+1032) (BuildId: a64ec0d2e59c0dc84e59dad386f412d8)
      #50 pc 0000000000140468  /apex/com.android.runtime/lib64/libart.so (art_quick_to_interpreter_bridge+88) (BuildId: a64ec0d2e59c0dc84e59dad386f412d8)
      #51 pc 00000000023329c4  /system/framework/arm64/boot-framework.oat (com.android.internal.os.ZygoteInit.main+2100) (BuildId: 7f789458715e814d322792a9aa87472ab8caa0b8)
      #52 pc 00000000001375b8  /apex/com.android.runtime/lib64/libart.so (art_quick_invoke_static_stub+568) (BuildId: a64ec0d2e59c0dc84e59dad386f412d8)
      #53 pc 000000000014600c  /apex/com.android.runtime/lib64/libart.so (art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+276) (BuildId: a64ec0d2e59c0dc84e59dad386f412d8)
      #54 pc 00000000004b0e34  /apex/com.android.runtime/lib64/libart.so (art::(anonymous namespace)::InvokeWithArgArray(art::ScopedObjectAccessAlreadyRunnable const&, art::ArtMethod*, art::(anonymous namespace)::ArgArray*, art::JValue*, char const*)+104) (BuildId: a64ec0d2e59c0dc84e59dad386f412d8)
      #55 pc 00000000004b0a24  /apex/com.android.runtime/lib64/libart.so (art::InvokeWithVarArgs(art::ScopedObjectAccessAlreadyRunnable const&, _jobject*, _jmethodID*, std::__va_list)+408) (BuildId: a64ec0d2e59c0dc84e59dad386f412d8)
      #56 pc 00000000003bb6b4  /apex/com.android.runtime/lib64/libart.so (art::JNI::CallStaticVoidMethodV(_JNIEnv*, _jclass*, _jmethodID*, std::__va_list)+624) (BuildId: a64ec0d2e59c0dc84e59dad386f412d8)
      #57 pc 00000000000c19ec  /system/lib64/libandroid_runtime.so (_JNIEnv::CallStaticVoidMethod(_jclass*, _jmethodID*, ...)+116) (BuildId: 58119cd499c9126f89ff7f6888912397)
      #58 pc 00000000000c49b0  /system/lib64/libandroid_runtime.so (android::AndroidRuntime::start(char const*, android::Vector<android::String8> const&, bool)+944) (BuildId: 58119cd499c9126f89ff7f6888912397)
      #59 pc 00000000000035bc  /system/bin/app_process64 (main+1388) (BuildId: ce48e0d699607c7f61196c3c69ba2920)
      #60 pc 000000000007e898  /apex/com.android.runtime/lib64/bionic/libc.so (__libc_init+108) (BuildId: 328a3518945f3e056fc055f1eea1cc35)
Lost connection to device.

**Comment by Imperial-lord** [comment]
Did you try this in an emulator? What is the behaviour there? Take a look here once - https://github.com/mapbox/mapbox-navigation-android/issues/438
